### PR TITLE
gcp_authn: polish the gcp_authn doc

### PR DESCRIPTION
**Pull request by tyxia**
The original configuration in the doc is from the integration test with fake server, which might be a bit confusing. See #24269.
Thus, update the configuration with production use case.

Signed-off-by: tyxia <tyxia@google.com>

### CODE REVIEW

File: docs/root/configuration/http/http_filters/_include/gcp-authn-filter-configuration.yaml [hunk 1]
```diff
@@ -57,20 +57,19 @@ static_resources:
...
-  # Cluster for fake metadata server.
+  # Cluster for GCE metadata server
   - name: gcp_authn
-    load_assignment:
-      cluster_name: gcp_authn
+    type: STRICT_DNS
+    connect_timeout
+      seconds: 5
+    dns_lookup_family: V4_ONLY
+    load_assignment {
```
Comment (a1lu) [inline docs/root/configuration/http/http_filters/_include/gcp-authn-filter-configuration.yaml:66]:
```suggestion
    load_assignment:
```

**Comment by tyxia** [comment]
/assign @phlax

Could you please take a look at this PR? Thanks!

**Review by phlax — COMMENTED** [review]
@tyxia iiuc this change to the docs is to use:

- an actual GCP authn server
- which is HTTP1 

from a docs pov the changes seem fine - im just a little apprehensive about a `google.internal` example address - is the only configurable address ?

**Comment by a1lu** [comment]
@phlax The google docs state this metadata server as the only way to query identity tokens, if you cannot use the gcloud SDK.
See:
https://cloud.google.com/run/docs/securing/service-identity#identity-tokens
https://cloud.google.com/run/docs/container-contract#metadata-server

**Comment by phlax** [comment]
thanks for explanation @a1lu - if this is the publicly documented way to configure this im fine with that

**Comment by tyxia** [comment]
> @tyxia iiuc this change to the docs is to use:
> 
> * an actual GCP authn server
> * which is HTTP1
> 
> from a docs pov the changes seem fine - im just a little apprehensive about a `google.internal` example address - is the only configurable address ?
Currently yes, as stated in gcp_authn [doc](https://github.com/tyxia/envoy/blob/ed2559e74545ff2f3cee9ec690fc57b5a5798db5/docs/root/configuration/http/http_filters/gcp_authn_filter.rst#gcp-authentication-filter), this filter is used to fetch authentication tokens from [Google Compute Engine(GCE) metadata server](https://cloud.google.com/compute/docs/metadata/overview). In the future, i could imagine there will have other addresses for metadata server but still will be gcp environment.

**Comment by phlax** [comment]
@tyxia seems like you will need a `main` merge - ci is a bit in flux atm - apologies for incovenience

**Pull request merged**